### DISCUSSION TITLE
Multiple probe locations/functions

### DISCUSSION THREAD

**Discussion by axg20202** [terjeio/ioSender#199] [Q&A]
I have a moveable probe that can be placed on the corner of the workpiece to set height and workpiece edge/corner offsets, but I also have a tool height Z probe, which is to be mounted in a fixed position. I'd like to be able to use both in different situations. The first for workpiece setup, and the second for tool changes only. The two probes will share the same probe input on the controller (grblHAL teensy) but obviously have different offset values etc. What's the best way to manage this in IOSender and during jobs in GCODE?

**Comment by terjeio** [comment]
[This grblHAL plugin](https://github.com/grblHAL/Templates/blob/master/my_plugin/probe%20select/my_plugin.c) would solve that?

**Reply by axg20202** [reply]
I think that assumes the different probes have dedicated pins. I'm currently limited to using the same pin for both probes. I did wonder if this requires a gcode macro to trigger different probing routines?

**Reply by terjeio** [reply]
The plugin uses a relay to switch between the probes.

**Reply by axg20202** [reply]
OK, thanks. I wonder if it could be simpllified so there is no need for that. It is only the location info and movements that are different, so surely they could just share the same connection?

**Reply by terjeio** [reply]
Simplest would be to use a manual switch. Not sure what you mean by the location and movements beeing different - and the probe connection is the same.
G59.3 is a location you set if you opt for automatic switching.

**Reply by axg20202** [reply]
My thinking is that they can share a single pin as they aren't being used at the same time - either can close the cicruit to register a probe contact. The difference is that the probing will be at different locations depending on which prob I wish to use, so the probing routine to move the tool to the probing point will be unique to each probe.

**Reply by terjeio** [reply]
> My thinking is that they can share a single pin as they aren't being used at the same time

Thats what the relay is for, to switch between two probes connected to the same pin. You can either use G28, G30 or one or more of the G5x offsets to store the probing points. With the plugin you can have automatically probe switching when within the G59.3 target.

You can always write your own plugin or modify the driver code if you want something else.

**Reply by axg20202** [reply]
OK, thanks, I will look into the GCODE options.  I'm still not sure I understand why a relay would be needed. Surely two NO switches/probes can be wired in parallel to the same signal pin, so closing either probe grounds the signal pin and generates a probe event.

**Comment by terjeio** [comment]
> Surely two NO switches/probes can be wired in parallel to the same signal pin

Yes they can. If this it is about moving to predefined positions then G28 and G30 is likely what you want to use. If probing from the probing tab in ioSender you can have different profiles for your probes - probe diameter, XY offsets etc.

**Reply by mintracer** [reply]
where are those profiles stored? i´m trying to save them, but nothing is saved... i´m using the latest XL 2.0.43

thx

**Reply by terjeio** [reply]
In _ProbingProfiles.xml_ in the sender folder. You have to give the profile a name before selecting _Add_ from the menu.

**Reply by mintracer** [reply]
https://github.com/terjeio/ioSender/assets/97637453/7948157b-4d90-4000-ada7-59a4f66a22e5

i downloaded a fresh copy from the latest release... i recorded a video... maybe it helps ... :) i´m on Windows 11... maybe that´s a problem... i generates the file, but it´s not edited...

thx

**Reply by terjeio** [reply]
Translation interfered with menu handling, please try the [latest edge version](http://www.io-engineering.com/downloads).

**Reply by mintracer** [reply]
2.0.44p2 works like a charm now! thx
